From 072c411860042644524263b6a4abe88a583802b9 Mon Sep 17 00:00:00 2001
From: Stefan Seyfried <seife@tuxbox-git.slipkontur.de>
Date: Fri, 1 Jan 2016 23:20:03 +0100
Subject: [PATCH 8004/8016] opkg_manager: do not read package list twice on
 start

checkUpdates() already gets the package list and the list of
upgradables, so it is unnecessary to do this again immediately
afterwards.

Signed-off-by: MarkusVolk <f_l_k@t-online.de>
---
 src/gui/opkg_manager.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/gui/opkg_manager.cpp b/src/gui/opkg_manager.cpp
index 79b951d..1fa1e94 100644
--- a/src/gui/opkg_manager.cpp
+++ b/src/gui/opkg_manager.cpp
@@ -468,7 +468,9 @@ bool COPKGManager::checkUpdates(const std::string & package_name, bool show_prog
 		status.hide();
 	}
 
+#if 0
 	pkg_map.clear();
+#endif
 
 	return ret;
 }
@@ -497,8 +499,10 @@ int COPKGManager::showMenu()
 	if (checkUpdates())
 		DisplayInfoMessage(g_Locale->getText(LOCALE_OPKG_MESSAGEBOX_UPDATES_AVAILABLE));
 
+#if 0
 	getPkgData(OM_LIST);
 	getPkgData(OM_LIST_UPGRADEABLE);
+#endif
 
 	menu = new CMenuWidget(g_Locale->getText(LOCALE_SERVICEMENU_UPDATE), NEUTRINO_ICON_UPDATE, width, MN_WIDGET_ID_SOFTWAREUPDATE);
 	menu->addIntroItems(LOCALE_OPKG_TITLE, NONEXISTANT_LOCALE, CMenuWidget::BTN_TYPE_BACK, CMenuWidget::BRIEF_HINT_YES);
-- 
2.1.4

